app/models/article_category.rb

```diff
@@ -0,0 +1,4 @@
+class ArticleCategory < ApplicationRecord
```

Comment:
Why need a model?

Reply:
The model here is used for linking the articles table and categories table together in a many-to-many relationship. It tracks the primary keys of both of these tables for each of the association.

Reply:
You didn't answer the question.
Try remove this file and make the following changes in your model & see if it still work as you expect

```
Category:
has_and_belongs_to_many :articles

Article:
has_and_belongs_to_many :categories
```

Reply:
Yep that will work as well in this particular case. By having a separate model for `ArticleCategory` allows for extending the association within the join table like validations or extra attributes if need be in the future.

Reply:
That should ideally still work as well in this particular case. However after testing it out, it seems to be looking for the `articles_categories` table instead of the `article_categories` with the previous association.  

To answer your earlier question, by having a separate model for `ArticleCategory` allows for extending the association within the join table like validations or extra attributes if need be in the future.

Reply:
1. This means your table naming has an issue.
2. Good